apps/s_time.c

```diff
@@ -155,6 +165,10 @@ int s_time_main(int argc, char **argv)
             break;
         case OPT_VERIFY:
             verify_args.depth = opt_int_arg();
+            if (verify_args.depth >= 0) {
+                X509_VERIFY_PARAM_set_depth(vpm, verify_args.depth);
```

Comment:
Wouldn't it be better to create the `vpm` via `X509_VERIFY_PARAM_new()` here rather than always as you do above. We could then eliminate `vpmtouched` and call `SSL_CTX_set1_param` if `vpm != NULL`